orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/ExecutionLatch.kt

```diff
@@ -46,6 +46,18 @@ class ExecutionLatch(private val predicate: Predicate<ExecutionComplete>) :
   fun await() = latch.await(10, TimeUnit.SECONDS)
 }
 
+fun ConfigurableApplicationContext.run(execution: PipelineExecution, launcher: (PipelineExecution) -> Unit) {
```

Comment:
dumb question: what's the purpose of this function?

Reply:
`runUntilCompletion()` starts the pipeline and then waits 10 seconds for it to finish, which doesn't work for the purposes of this test when we need to mutate the queue after starting the execution.

This method is the same as `runUntilCompletion` without the part that waits. The sleep is to allow the messages for starting the execution to be processed before we continue on to deleting the queue.